File: src/dataset/python/word-count_exercism-ThomasZumsteg/claude/ClaudeSonnet4_wordcount_v3.py
"""Catalogs and counts the number of words used in a phrase"""
from collections import Counter
import re

_WORD_PATTERN = re.compile(r'[\W_]+', re.UNICODE)

def word_count(phrase):
    """Returns dictionary with a count of words used"""
    if not phrase:
        return Counter()
    
    words = _WORD_PATTERN.split(phrase.lower())
    return Counter(word for word in words if word)

File: src/dataset/javascript/word-count_exercism-ThomasZumsteg/claude/ClaudeSonnet4_word-count_v3.js
var Words = function() {}

Words.prototype.count = function( wordString ) {
	var wordCount = {};
	var trimmed = wordString.trim();
	
	if (!trimmed) return wordCount;
	
	var words = trimmed.split(/\s+/);
	var length = words.length;
	
	for (var i = 0; i < length; i++) {
		var word = words[i];
		wordCount[word] = (wordCount[word] || 0) + 1;
	}
	
	return wordCount;
};

module.exports = Words;